### PR TITLE
Add feedback feature #161

### PR DESCRIPTION
**Pull request by kerollaine**
I was not able to fully test the l10n/i18n on my development machine (for some reason LOCALE_PATHS is not working). I could use some help testing that (I did add the corresponding new keys in the translation files).

Other than that, the feature is fully functional. By default, the e-mails are being sent to feedback@speakerfight.com, properly configured to instruct your e-mail client to reply automatically to the logged user e-mail address. See the test cases for more info.

### CODE REVIEW

**Comment by luanfonceca** [comment]
Can you add a print?

**Comment by luanfonceca** [comment]
About the email, its okay to use that one, the only concern is because he doesnt exist, o didnt setup a smtp server yet, but in the server i can use the `local_settings.py` and setup my own email.

o/

**Comment by kerollaine** [comment]
Yep! Here they are:

1.

feedback-1

2.

feedback2

3.

feedback3

**Comment by luanfonceca** [comment]
Nice, as soon as you finish the pending changes that i ask on the Review, i can merge and deploy you contribution.

o/

File: core/urls.py [hunk 1]
```diff
@@ -23,4 +23,5 @@
         view=views.ProfileUpdateView.as_view(),
         name='update_profile'),
     url(r'^jsi18n/$', javascript_catalog, name='javascript_catalog'),
+    url(r'/feedback/', views.FeedbackView.as_view(), name='feedback'),
```
Comment (mauricioabreu) [inline core/urls.py]: I think it should be `^feedback/$`.

File: core/views.py [hunk 2]
```diff
@@ -85,3 +87,18 @@ def form_invalid(self, form):
...
 
         return self.get()
+
+
+class FeedbackView(LoginRequiredMixin, FormView):
+    template_name = 'feedback.html'
+    form_class = FeedbackForm
+
+    def form_valid(self, form):
+        message = mail.EmailMessage(form.cleaned_data['subject'],
```
Comment (mauricioabreu) [inline core/views.py]: To cause less confusion here, you may wanna change `message` to `email` or `email_message`.

File: core/views.py [hunk 3]
```diff
@@ -85,3 +87,18 @@ def form_invalid(self, form):
...
+                                    form.cleaned_data['message'],
+                                    settings.NO_REPLY_EMAIL,
+                                    [settings.FEEDBACK_EMAIL],
+                                    headers={'Reply-To': self.request.user.email})
+        message.send()
+        messages.success(self.request, _('Thank you for your feedback! We\'ll get in touch as soon as possible.'))
```
Comment (mauricioabreu) [inline core/views.py]: To avoid the escape you can have this string using double quotes. :)

File: core/views.py [hunk 4]
```diff
@@ -85,3 +87,18 @@ def form_invalid(self, form):
...
+        return HttpResponseRedirect('/events/')
```
Comment (mauricioabreu) [inline core/views.py]: Here you can use the Django's reverse module. This way if we change this URL it won't raise errors.

File: speakerfight/settings.py [hunk 5]
```diff
@@ -40,6 +40,7 @@
 
 STATIC_ROOT = os.path.join(BASE_DIR, "static")
 DEFAULT_FROM_EMAIL = NO_REPLY_EMAIL = 'Speakerfight <noreply@speakerfight.com>'
+FEEDBACK_EMAIL = 'Speakerfight <feedback@speakerfight.com>'
```
Comment (mauricioabreu) [inline speakerfight/settings.py]: Can't we use the same email (DEFAULT_FROM_EMAIL)?

**Comment by mauricioabreu** [comment]
Despite the changes I asked, you did a very good job here. :)

File: core/templates/feedback.html [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+{% endblock head %}
+
+{% block content %}
+  <div class="row header">
+    <h2>
+      {% trans "Feedback" %}
+    </h2>
+  </div>
+  <div class="row form-wrapper">
+    <div class="column">
```
Comment (victor-torres) [inline core/templates/feedback.html]: You could add a message somewhere in this template with something like:

> You can also chat with us on Gitter.im

What do you think?

File: speakerfight/settings.py (hunk 5 above)
Comment (kerollaine) [inline speakerfight/settings.py]: Yes, it's possible. However, I think that the 'DEFAULT_FROM_EMAIL' setting has not the same goal as 'FEEDBACK_EMAIL'. Note that the former is commonly used to discourage the user to reply any contact, while the latter should encourage continuous communication. This is the rationale came with when developing the feature.

Comment (victor-torres) [inline speakerfight/settings.py]: Makes sense, @kerollaine, but if this e-mail address will only notify the webmaster (do we use this term yet? haha) about users’ feedback then there’s no need to receive responses and we can stuck with noreply@[...].

File: core/templates/feedback.html [hunk 7]
```diff
@@ -40,6 +40,7 @@
           </button>
         {% endbuttons %}
       </form>
+      <p>{% trans 'You can also chat with us on' %}<a target="_blank" href="https://gitter.im/luanfonceca/speakerfight"> Gitter.im</a></p>
```
Comment (victor-torres) [inline core/templates/feedback.html]: 👍

File: speakerfight/settings.py (hunk 5 above)
Comment (mauricioabreu) [inline speakerfight/settings.py]: @kerollaine your argument is plausible. Cool!

File: core/urls.py [hunk 8]
```diff
@@ -23,4 +23,5 @@
         view=views.ProfileUpdateView.as_view(),
         name='update_profile'),
     url(r'^jsi18n/$', javascript_catalog, name='javascript_catalog'),
+    url(r'/feedback/$', views.FeedbackView.as_view(), name='feedback'),
```
Comment (mauricioabreu) [inline core/urls.py]: I think you want `^/feedback/$`

Comment (mauricioabreu) [inline core/urls.py]: Yep! I am not sure what I tried to say that day but I guess I tried to say the same thing the docs say here:
https://docs.djangoproject.com/pt-br/1.10/topics/http/urls/
URLs don't have the first `/`.

**Pull request closed without merging**